Kong/kuma-gui
# Dataplane Wizard

This is the first of 2 stages for the Dataplane Wizard. The first stage being the Kubernetes flow, and the second being Universal.

[Here is a quick video recording of it in action](https://share.getcloudapp.com/7Ku0yYd4)

### To test it:
1. Start up the GUI: `yarn install` (if you haven't already), and `yarn serve`
2. **For Kubernetes:** http://localhost:8080/#/wizard/kubernetes-dataplane / **For Universal:** http://localhost:8080/#/wizard/universal-dataplane
3. Go through the form and test each option available to see conditional handling and display

### Notes:
- The last step does not yet display code for creating a Dataplane. That part is still in progress.
- You will see a panel at the start of this wizard that will detect what environment Kuma is running in, and provide instructions to view the other environment's wizard.
- You'll see a "Debugger" panel in the right sidebar. As noted in the component itself, this is only displayed when running the GUI in development mode. The app runs in dev mode when using `yarn serve` in this repo, but runs in production mode when compiled and running from within Kuma. This panel exists simply to keep track of `v-model` data that will be used to assemble the code output.
